File: OOP_final_project_JAVA_version/src/exercicio_incremental_parte2/Veiculo.java
package exercicio_incremental_parte2;

import java.util.Iterator;

public class Veiculo {
    private String placa;
    private int ano;
    private String modelo;
    private int capacidade;
    private int tipo;
    private int num_contrato;
    private Contrato contrato;

    public Veiculo(String placa, int ano, String modelo, int capacidade, int tipo, int num_contrato) {
        this.placa = placa;
        this.ano = ano;
        this.modelo = modelo;
        this.capacidade = capacidade;
        this.tipo = tipo;
        this.num_contrato = num_contrato;
        //exigeContrato();
    }

    public Veiculo(String placa, int ano, String modelo, int capacidade, int tipo) {
        this.placa = placa;
        this.ano = ano;
        this.modelo = modelo;
        this.capacidade = capacidade;
        this.tipo = tipo;
        exigeContrato();
    }
    
    public void exigeContrato(){
        if(this.getTipo()==1 && this.getNum_contrato()==0){
            System.out.println("insira o numero do contrato");
        }else{
            this.num_contrato = num_contrato;
        }
    }
    
    public void verificaVeiculo(){
     if(this.getTipo() == 0){
        String t = "o veiculo é proprio";
        System.out.println(t);
         //return t;
     }else{
        String t = "o veiculo é alugado";
        System.out.println(t);
        //return t;
     }
    }
    
    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public int getAnod() {
        return ano;
    }

    public void setAno(int ano) {
        this.ano = ano;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public int getCapacidade() {
        return capacidade;
    }

    public void setCapacidade(int capacidade) {
        this.capacidade = capacidade;
    }

    public int getTipo() {
        return tipo;
    }

    public void setTipo(int tipo) {
        this.tipo = tipo;
    }

    public int getNum_contrato() {
        return num_contrato;
    }

    public void setNum_contrato(int num_contrato) {
        if(this.getTipo()==0){
            System.out.println("Método inválido");
            System.out.println("Nao se pode colocar numero de contrato pois o veiculo é próprio");
        }else{
            this.num_contrato = num_contrato;
        }     
    }
    
    //public void teste(){
       // this.setNum_contrato(4);
    //}

    Iterator<Veiculo> iterator() {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    void remove(Veiculo veiculo) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    void add(Veiculo veiculo) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }
}

File: OOP_final_project_python_version/veiculo.py
class Veiculo:
    __total_veiculos = 0

    def __init__(self, placa, ano, modelo, capacidade, tipo, num_contrato=None):
        self.__placa = placa
        self.__ano = ano
        self.__modelo = modelo
        self.__capacidade = capacidade
        self.__tipo = tipo
        self.__num_contrato = num_contrato
        Veiculo.__total_veiculos += 1
        self.__num_veiculo = Veiculo.__total_veiculos

    @property
    def placa(self):
        return self.__placa

    @placa.setter
    def placa(self, placa):
        self.__placa = placa

    @property
    def ano(self):
        return self.__ano

    @ano.setter
    def ano(self, ano):
        self.__ano = ano

    @property
    def modelo(self):
        return self.__modelo

    @modelo.setter
    def modelo(self, modelo):
        self.__modelo = modelo

    @property
    def capacidade(self):
        return self.__capacidade

    @capacidade.setter
    def capacidade(self, capacidade):
        self.__capacidade = capacidade

    @property
    def tipo(self):
        return self.__tipo

    @tipo.setter
    def tipo(self, tipo):
        self.__tipo = tipo

    @property
    def num_contrato(self):
        return self.__num_contrato

    @num_contrato.setter
    def num_contrato(self, num_contrato):
        if self.tipo == 0:
            print("Método inválido")
            print("Não é possível colocar número de contrato porque o veículo é próprio.")
        else:
            self.__num_contrato = num_contrato

    @property
    def num_veiculo(self):
        return self.__num_veiculo

    def exige_contrato(self):
        if self.tipo == 1 and self.num_contrato is None:
            print("Insira o número do contrato.")
        else:
            self.num_contrato = self.num_contrato

    def verifica_veiculo(self):
        if self.tipo == 0:
            print("O veículo é próprio.")
        else:
            print("O veículo é alugado.")
